src/components/document/mywork-document-or-browser.tsx

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import { useAppConfig } from "../../hooks/use-stores";
+import { DocumentModelType } from "../../models/document/document";
+import { ENavTab } from "../../models/view/nav-tabs";
+import { SubTabsPanel } from "../navigation/sub-tabs-panel";
+import { DocumentCollectionList } from "../thumbnail/document-collection-list";
+import { EditableDocumentContent, IProps as IEditableDocumentContentProps } from "./editable-document-content";
+
+interface IProps extends IEditableDocumentContentProps {
+  showBrowser: boolean;
+  onSelectNewDocument?: (type: string) => void;
+  onSelectDocument?: (document: DocumentModelType) => void;
+}
+
+export const MyWorkDocumentOrBrowser: React.FC<IProps> = props => {
+  const navTabs = useAppConfig().navTabs;
+  const myWorkTabSpec = navTabs.getNavTabSpec(ENavTab.kMyWork);
+  const { showBrowser, document, onSelectNewDocument, onSelectDocument, ...others } = props;
+
+  if (showBrowser && myWorkTabSpec) {
+    // If we are re-rendered and showBrowser is true then we will will always
+    // re-render the SubTabsPanel because of the renderSubTabPanel property.
+    // However since we are passing most of our props to DocumentCollectionList
+    // it is likely that every re-render will be because one of these props
+    // changed and we need to re-render the SubTabsPanel anyhow.
+    return (
+      <SubTabsPanel tabSpec={myWorkTabSpec} renderSubTabPanel={(subTab) => (
+        <div>
+          <DocumentCollectionList
+            subTab={subTab}
+            tabSpec={myWorkTabSpec}
+            onSelectNewDocument={onSelectNewDocument}
+            onSelectDocument={onSelectDocument}
+          />
+        </div>
+      )} />
+    );
```

Comment:
This is the common way to express React render props. We don't use them much, so it looks foreign, but I'm inclined to keep it this way to match other examples of render properties. Here are some:
https://legacy.reactjs.org/docs/render-props.html
https://itnext.io/best-practices-of-render-props-pattern-to-improve-react-performance-and-more-part-1%EF%B8%8F%E2%83%A3-fb8e9b3e9eec

You could also think of it as an anonymous functional component, but that isn't quite right because SubTabsPanel isn't using JSX or calling `React.createElement` to make an element out of the passed in function. I think making an element is how React adds support for things like useState and useEffect to functional components.